pkg/installmanager/installmanager_test.go

```diff
@@ -717,3 +722,72 @@ func Test_pasteInPullSecret(t *testing.T) {
 		})
 	}
 }
+
+func TestPatchWorkerMachineSet(t *testing.T) {
```

Comment:
This is a neat test, but I feel like we should be able to go higher-level and not care about the underlying implementation as much.

To facilitate that, we could mock the various ioutils to spoof walking the file system...

But I recall that's kind of a pain. I'd be okay if we just pulled out the chunk of code that starts with a MachinePool and a single file already loaded in a []byte, and ends with that []byte (possibly) having been updated to include the new filter. The way you're unmarshalling that as a MachineSet to introspect it is great.